File: bindings/clojure/bench/dev/everanium/itb/clojure/bench_message.clj
(ns dev.everanium.itb.clojure.bench-message
  "encrypt-message throughput vs plaintext size (Single Message
  profile) at 1 MiB / 16 MiB / 64 MiB."
  (:require [dev.everanium.itb.clojure.bench-util :as u]
            [dev.everanium.itb.clojure.core :as itb]))

(defn run []
  (with-open [pipe (itb/init (u/profile-name "singlemsg-triple-nomac-v1")
                             (u/build-opts))]
    (u/header)
    (doseq [size u/sizes]
      (let [plain (byte-array size)]
        (u/csprng-fill plain)
        (u/bench-case "message" size #(itb/encrypt-message pipe plain))))))

(defn -main [& _]
  (u/apply-runtime-caps!)
  (run)
  (System/exit 0))

File: bindings/vbnet/bench/EveraniumItb.VisualBasic.Bench/BenchMessage.vb
' EncryptMessage throughput vs plaintext size (Single Message
' profile) at 1 MiB / 16 MiB / 64 MiB.

Imports System.Security.Cryptography
Imports Everanium.Itb.VisualBasic

Friend Module BenchMessage

    Friend Sub Run()
        Using pipe As Pipeline = Pipeline.Init(
                ProfileName("singlemsg-triple-nomac-v1"), BuildOpts())
            Header()
            For Each size As Integer In Sizes
                Dim plain(size - 1) As Byte
                ' CSPRNG-fill so plaintext content matches the root Go
                ' bench (crypto/rand). Not in the timing loop.
                RandomNumberGenerator.Fill(plain)
                [Case]("message", size, Sub() pipe.EncryptMessage(plain))
            Next
        End Using
    End Sub
End Module
